client/handle_http.go

```diff
@@ -2256,6 +2243,82 @@ func runPut(request *http.Request, responseChan chan<- *http.Response, errorChan
 
 }
 
+// This function queries the director with a PROPFIND to attempt to get the 'dirListHost'. If a propfind is not allowed on the director
+// We fall back to the deprecated dirlisthost in the namespace
+func getDirListHost(ctx context.Context, remoteObjectUrl *url.URL, namespace namespaces.Namespace, directorUrl string) (dirListHost *url.URL, err error) {
+	// If we are a recursive download and using the director, we want to attempt to get directory listings from
+	// PROPFINDing the director
+	if directorUrl != "" {
+		// Query the director a PROPFIND to see if we can get our directory listing
+		resp, err := queryDirector(ctx, "PROPFIND", remoteObjectUrl.Path, directorUrl)
+		if err != nil {
+			// If we have an issue querying the director, we want to fallback to the deprecated dirlisthost from the namespace
+			// At this point, we have already queried the director (and it should have succeeded if we are here) so the error
+			// we get is most likely an issue with PROPFIND (and an outdated director). Therefore, we should continue if we get a 404
+			if resp == nil || !(resp.StatusCode == http.StatusNotFound) {
+				// We did not get a 404 so we should return the error
+				return nil, err
+			}
+		}
+		if resp.StatusCode == http.StatusMethodNotAllowed || resp.StatusCode == http.StatusNotFound {
+			// If the director responds with 405 (method not allowed) or a 404 (not found), we're working with an old Director.
+			// In that event, we try to fallback and use the deprecated dirlisthost
+			log.Debugln("Failed to find collections url from director, attempting to find directory listings host in namespace")
+			// Check for a dir list host in namespace
+			if namespace.DirListHost == "" {
+				// Both methods to get directory listings failed
+				err = &dirListingNotSupportedError{Err: errors.New("origin and/or namespace does not support directory listings")}
+				return nil, err
+			} else {
+				dirListHost, err = url.Parse(namespace.DirListHost)
+				if err != nil {
+					return nil, err
+				}
+				return dirListHost, nil
+			}
+		} else if resp.StatusCode == http.StatusTemporaryRedirect {
+			// If the director responds with a 307 (temporary redirect), we're working with a new Director.
+			// In that event, we can get the collections URL from our redirect
+			collections := resp.Header.Get("Location")
```

Comment:
Why get this from the `Location` header instead of the `X-Pelican-Namespace`'s built-in `collections-url`?

Reply:
I need to look through more to remember why, I think it is because the `collections-url` is set from the old school `dirlisthost` being set in topology. Since we want to move away from that, we just look directly at the `Location` header to get the origin (if listings are enabled) rather than relying on dirlisthost being set